server/datastore/mysql/apple_mdm.go

```diff
@@ -522,8 +533,8 @@ func upsertMDMAppleHostMDMInfoDB(ctx context.Context, tx sqlx.ExtContext, server
 
 	args := []interface{}{}
 	parts := []string{}
-	for _, h := range hosts {
-		args = append(args, enrolled, serverURL, fromSync, mdmID, false, h.ID)
+	for _, id := range hostIDs {
+		args = append(args, enrolled, serverURL, fromSync, mdmID, false, id)
```

Comment:
Do we have a rough idea of how many host IDs we may get in a single call? Because if that's in the thousands, we should batch those INSERTs to avoid any buffer size limits and such.

Reply:
Answering my own question, I think those are the devices to sync or fetch (https://developer.apple.com/documentation/devicemanagement/syncdevicerequest) so it's 100 by default, 1000 at most if we change the limit (I don't think we do atm in the cron ingestion job). So that should be fine.